imx.mk

```diff
@@ -175,22 +176,66 @@ $(ROOT)/out/boot.scr: $(ROOT)/build/imx/u-boot_boot_script | $(ROOT)/out
 	$(U-BOOT_PATH)/tools/mkimage -T script -C none -n 'Boot script' \
 		-d $< $@
 
+################################################################################
+# Flash images
+################################################################################
+USE_PERSISTENT_ROOTFS ?= 0
+
+# Configuration of the BOOT partition
+FLASH_PARTITIONS_BLOCK_SIZE = 512
+FLASH_PARTITION_BOOT_START_BLOCK = 16384
+FLASH_PARTITION_BOOT_SIZE_IN_BYTES = \
+	$(shell echo $$(( 64 * 1024 * 1024 )))
+FLASH_PARTITION_BOOT_SIZE_IN_BLOCKS = \
+	$(shell echo $$(( $(FLASH_PARTITION_BOOT_SIZE_IN_BYTES) / $(FLASH_PARTITIONS_BLOCK_SIZE) )))
+FLASH_PARTITIONS_TABLE = "\
+	start=$(FLASH_PARTITION_BOOT_START_BLOCK) \
+	size=$(FLASH_PARTITION_BOOT_SIZE_IN_BLOCKS) \
+	type=7\n"
+FLASH_IMAGE_SIZE = \
+	$(shell echo $$(( $(FLASH_PARTITION_BOOT_START_BLOCK) * $(FLASH_PARTITIONS_BLOCK_SIZE) \
+	+ $(FLASH_PARTITION_BOOT_SIZE_IN_BYTES) )))
+
+# Configuration of the ROOTFS partition if enabled
+ifeq ($(USE_PERSISTENT_ROOTFS),1)
+FLASH_PARTITION_ROOTFS_IMAGE_PATH = $(ROOT)/out-br/images/rootfs.ext2
+FLASH_PARTITION_ROOTFS_START_BLOCK = \
+	$(shell echo $$(( $(FLASH_PARTITION_BOOT_START_BLOCK) + $(FLASH_PARTITION_BOOT_SIZE_IN_BLOCKS) )))
+FLASH_PARTITION_ROOTFS_SIZE_IN_BYTES = \
+	$(shell stat --printf="%s" $(FLASH_PARTITION_ROOTFS_IMAGE_PATH))
+FLASH_PARTITION_ROOTFS_SIZE_IN_BLOCKS = \
+	$(shell echo $$(( $(FLASH_PARTITION_ROOTFS_SIZE_IN_BYTES) / $(FLASH_PARTITIONS_BLOCK_SIZE) )))
+FLASH_PARTITIONS_TABLE += "\
+	start=$(FLASH_PARTITION_ROOTFS_START_BLOCK) \
+	size=$(FLASH_PARTITION_ROOTFS_SIZE_IN_BLOCKS) \
+	type=83\n"
+FLASH_IMAGE_SIZE := $(shell echo $$(( $(FLASH_IMAGE_SIZE) + $(FLASH_PARTITION_ROOTFS_SIZE_IN_BYTES) )))
+endif
+
 .PHONY: flash-image
 flash-image: buildroot mkimage
 	$(MAKE) flash-image-only
 
 .PHONY: flash-image-only
-flash-image-only: $(ROOT)/out-br/images/ramdisk.img $(ROOT)/out/boot.scr
```

Comment:
We should still depend on `$(ROOT)/out-br/images/ramdisk.img` in case `USE_PERSISTENT_ROOTFS` ==  `0`